File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{   
    int coins = 0;
    float change;
    int cents;
    
    do
    {
        printf("Oh hai! How much change is owed?");
    
        change = GetFloat() * 100;
    }
    while(change <= 0);
    
    cents = round(change);
    
    if(cents >= 25)
    {
        coins = coins + (cents / 25);
        cents = cents % 25;
    }
    
    if(cents >= 10)
    {
        coins = coins + (cents / 10);
        cents = cents % 10;
    }
    
    if(cents >= 5)
    {
        coins = coins + (cents / 5);
        cents = cents % 5;
    }
    
    if(cents >= 1)
    {
        coins = coins + (cents / 1);
        cents = cents % 1;
    }
    
    printf("%i\n", coins);
    
}

File: pset6/greedy.py
import math

def main():
    coins = 0
    change = math.floor(float(input("Oh hai! How much change is owed?")) * 100)
    
    while change <= 0:
        change = math.floor(float(input("Oh hai! How much change is owed?")) * 100)
        
    while change != 0:
        if change >= 25:
            coins += 1
            change -= 25
            
        elif change >= 10:
            coins  += 1
            change -= 10
            
        elif change >= 5:
            coins += 1
            change -= 5
            
        elif change >= 1:
            coins += 1
            change -= 1
            
    print(coins)
    

if __name__ == '__main__':
    main()